Except on aarch64-darwin the build of `taskflow` appears to be trying to use x86 intrinsics:

```
In file included from In file included from /nix/store/nwfcj1p4l74krm4n6szrb0c534p4ai2s-clang-wrapper-11.1.0/resource-root/include/immintrin.h/tmp/nix-build-taskflow-3.3.0.drv-0/source/taskflow/core/../utility/os.hpp::15126:
:
/nix/store/1r3mrri4afdhshaibaamb3rafvy4ngv9-libcxx-11.1.0-dev/include/c++/v1/math.h/nix/store/nwfcj1p4l74krm4n6szrb0c534p4ai2s-clang-wrapper-11.1.0/resource-root/include/mmintrin.h::64933::125:/nix/store/1r3mrri4afdhshaibaamb3rafvy4ngv9-libcxx->
'__builtin_isless' declared here
    return isless(__lcpp_x, __lcpp_y);
           ^    __builtin_ia32_emms();
In file included from /nix/store/nwfcj1p4l74krm4n6szrb0c534p4ai2s-clang-wrapper-11.1.0/resource-root/include/immintrin.h    ^:
15:
/nix/store/nwfcj1p4l74krm4n6szrb0c534p4ai2s-clang-wrapper-11.1.0/resource-root/include/mmintrin.h:33:5: fatal error: use of undeclared identifier '__builtin_ia32_emms'; did you mean '__builtin_isless'?
/nix/store/1r3mrri4afdhshaibaamb3rafvy4ngv9-libcxx-11.1.0-dev/include/c++/v1/math.h:649:In file included from 12:: /tmp/nix-build-taskflow-3.3.0.drv-0/source/examples/subflow.cpp :note: '__builtin_isless' declared here
```

--- a/pkgs/development/libraries/jarowinkler-cpp/default.nix
+++ b/pkgs/development/libraries/jarowinkler-cpp/default.nix
@@ -1,0 +1,41 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, cmake
+, catch2
+}:
+
+stdenv.mkDerivation rec {
+  pname = "jarowinkler-cpp";
+  version = "1.0.0";
+
+  src = fetchFromGitHub {
+    owner = "maxbachmann";
+    repo = "jarowinkler-cpp";
+    rev = "v${version}";
+    hash = "sha256-6dIyCyoPs/2wHyGqlE+NC0pwz5ggS5edhN4Jbltx0jg=";
+  };
+
+  nativeBuildInputs = [
+    cmake
+  ];
+
+  cmakeFlags = lib.optionals doCheck [
+    "-DRAPIDFUZZ_BUILD_TESTING=ON"
+  ];
+
+  checkInputs = [
+    catch2
+  ];
+
+  # uses unreleased Catch2 version 3
+  doCheck = false;
+
+  meta = {
+    description = "Fast Jaro and Jaro-Winkler distance";
+    homepage = "https://github.com/maxbachmann/jarowinkler-cpp";
+    license = lib.licenses.mit;
+    maintainers = with lib.maintainers; [ dotlambda ];
+    platforms = lib.platforms.unix;
+  };
+}

--- a/pkgs/development/libraries/rapidfuzz-cpp/default.nix
+++ b/pkgs/development/libraries/rapidfuzz-cpp/default.nix
@@ -1,0 +1,41 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, cmake
+, catch2
+}:
+
+stdenv.mkDerivation rec {
+  pname = "rapidfuzz-cpp";
+  version = "1.0.1";
+
+  src = fetchFromGitHub {
+    owner = "maxbachmann";
+    repo = "rapidfuzz-cpp";
+    rev = "v${version}";
+    hash = "sha256-331iW0nu5MlxuKNTgMkRSASnglxn+hEWBhRMnw0lY2Y=";
+  };
+
+  nativeBuildInputs = [
+    cmake
+  ];
+
+  cmakeFlags = lib.optionals doCheck [
+    "-DRAPIDFUZZ_BUILD_TESTING=ON"
+  ];
+
+  checkInputs = [
+    catch2
+  ];
+
+  # uses unreleased Catch2 version 3
+  doCheck = false;
+
+  meta = {
+    description = "Rapid fuzzy string matching in C++ using the Levenshtein Distance";
+    homepage = "https://github.com/maxbachmann/rapidfuzz-cpp";
+    license = lib.licenses.mit;
+    maintainers = with lib.maintainers; [ dotlambda ];
+    platforms = lib.platforms.unix;
+  };
+}

--- a/pkgs/development/libraries/taskflow/default.nix
+++ b/pkgs/development/libraries/taskflow/default.nix
@@ -1,0 +1,50 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, substituteAll
+, doctest
+, cmake
+}:
+
+stdenv.mkDerivation rec {
+  pname = "taskflow";
+  version = "3.3.0";
+
+  src = fetchFromGitHub {
+    owner = "taskflow";
+    repo = "taskflow";
+    rev = "v${version}";
+    hash = "sha256-UfXGupxgtowIt3BnIVWwim3rTE57TT1C9TCx9LVyN34=";
+  };
+
+  patches = [
+    (substituteAll {
+      src = ./unvendor-doctest.patch;
+      inherit doctest;
+    })
+  ];
+
+  postPatch = ''
+    rm -r 3rd-party
+
+    # tries to use x86 intrinsics on aarch64-darwin
+    sed -i '/^#if __has_include (<immintrin\.h>)/,/^#endif/d' taskflow/utility/os.hpp
+  '';
+
+  nativeBuildInputs = [
+    cmake
+  ];
+
+  doCheck = true;
+
+  meta = {
+    description = "General-purpose Parallel and Heterogeneous Task Programming System";
+    homepage = "https://taskflow.github.io/";
+    changelog = let
+      release = lib.replaceStrings ["."] ["-"] version;
+    in "https://taskflow.github.io/taskflow/release-${release}.html";
+    license = lib.licenses.mit;
+    platforms = lib.platforms.unix;
+    maintainers = with lib.maintainers; [ dotlambda ];
+  };
+}

--- a/pkgs/development/libraries/taskflow/unvendor-doctest.patch
+++ b/pkgs/development/libraries/taskflow/unvendor-doctest.patch
@@ -1,0 +1,21 @@
+diff --git a/unittests/CMakeLists.txt b/unittests/CMakeLists.txt
+index 3397d798..8277191e 100644
+--- a/unittests/CMakeLists.txt
++++ b/unittests/CMakeLists.txt
+@@ -1,6 +1,6 @@
+ enable_testing()
+ 
+-include(${TF_3RD_PARTY_DIR}/doctest/doctest.cmake)
++include(@doctest@/lib/cmake/doctest/doctest.cmake)
+ 
+ list(APPEND TF_UNITTESTS 
+   utility 
+@@ -24,7 +24,7 @@ list(APPEND TF_UNITTESTS
+ foreach(unittest IN LISTS TF_UNITTESTS)
+   add_executable(${unittest} ${unittest}.cpp)
+   target_link_libraries(${unittest} ${PROJECT_NAME} tf::default_settings)
+-  target_include_directories(${unittest} PRIVATE ${TF_3RD_PARTY_DIR}/doctest)
++  target_include_directories(${unittest} PRIVATE @doctest@/include/doctest)
+   doctest_discover_tests(${unittest})
+ endforeach()
+ 

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -20181,6 +20181,8 @@ with pkgs;
 
   rapidcheck = callPackage ../development/libraries/rapidcheck {};
 
+  rapidfuzz-cpp = callPackage ../development/libraries/rapidfuzz-cpp { };
+
   rapidjson = callPackage ../development/libraries/rapidjson {};
 
   rapidxml = callPackage ../development/libraries/rapidxml {};

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -17890,6 +17890,8 @@ with pkgs;
 
   jansson = callPackage ../development/libraries/jansson { };
 
+  jarowinkler-cpp = callPackage ../development/libraries/jarowinkler-cpp { };
+
   jasper = callPackage ../development/libraries/jasper { };
 
   jbig2dec = callPackage ../development/libraries/jbig2dec { };

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -20700,6 +20700,8 @@ with pkgs;
 
   tagparser = callPackage ../development/libraries/tagparser { };
 
+  taskflow = callPackage ../development/libraries/taskflow { };
+
   tclap = callPackage ../development/libraries/tclap {};
 
   tcllib = callPackage ../development/libraries/tcllib { };

